src/appleseed/renderer/modeling/camera/multilenscamera.cpp

```diff
@@ -0,0 +1,1328 @@
+            const double square_dist_film_lens = square_norm(film_point);
+            const double dist_film_lens = std::sqrt(square_dist_film_lens);
+            const double cos_theta = m_focal_length / dist_film_lens;
+            const double solid_angle = m_pixel_area * cos_theta / square_dist_film_lens;
+            importance = 1.0f / static_cast<float>(square_norm(outgoing) * solid_angle);
+
+            // The connection was possible.
+            return true;
+        }
+
+    private:
+        // Parameters.
+        double                   m_f_number;                 // F-number
+        bool                     m_autofocus_enabled;        // is autofocus enabled?
+        Vector2d                 m_autofocus_target;         // autofocus target on film plane, in NDC
+        double                   m_focal_distance;           // focal distance in camera space
+        bool                     m_diaphragm_map_bound;      // is a diaphragm map bound to the camera
+        size_t                   m_diaphragm_blade_count;    // number of blades of the diaphragm, 0 for round aperture
+        double                   m_diaphragm_tilt_angle;     // tilt angle of the diaphragm in radians
+        Source::Hints            m_diaphragm_map_hints;
+
+        bool                     m_derivatives_enabled;
+        std::vector<LensElement> m_lens_container;           // container of the lens elements
+        std::string              m_lens_file;                // path to file, where lens configuration is stored
+        int                      m_aperture_index;           // index of the aperture in the lens container
+        double                   m_lens_focal_length;        // original focal length of the lens
+        double                   m_lens_f_number;            // original F-number of the lens
+
+        double                   entrance_pupil_radius;      // radius of the entrance pupil
+        double                   entrance_pupil_center_z;    // z coordinate of the center of the entrance pupil
+        double                   exit_pupil_radius;          // list of radii of exit pupils from center to edge of film
+        double                   exit_pupil_center_z;        // z coordinate of the center of the exit pupil
+        double                   m_total_z;                  // z coordinate of the film plane
+        double                   m_last_z;                   // z coordinate of the last lens element
+
+        // Vertices of the diaphragm polygon.
+        std::vector<Vector2d>    m_diaphragm_vertices;
+        // Vertices of the diaphragm polygon translated to camera space coordinates.
+        std::vector<Vector2d>    m_diaphragm_vertices_camera;
+
+        // Importance sampler to sample the diaphragm map.
+        std::unique_ptr<ImageImportanceSamplerType>
+            m_importance_sampler;
+
+        void extract_lens_file()
+        {
+            if (has_param("lens_file"))
```

Comment:
What happens in the overall workflow, if the parameter doesn't exist? If you're sure it's always there, remove this check, if it may not exist, adapt the rest of the workflow.